File: backend/models/BehavioralLog.js
'use strict';
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const BehavioralLog = sequelize.define('BehavioralLog', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    session_id: { type: DataTypes.INTEGER, allowNull: false },
    event_type: {
      type: DataTypes.ENUM('gaze_away', 'head_movement', 'tab_switch', 'face_absent', 'multiple_persons'),
      allowNull: false
    },
    metadata: { type: DataTypes.JSON, allowNull: true },
    event_timestamp: { type: DataTypes.DATE, allowNull: false }
  }, {
    tableName: 'behavioral_logs',
    underscored: true
  });

  return BehavioralLog;
};

File: backend/app/models/behavioral_log.py
from datetime import datetime

from app.extensions import db


class BehavioralLog(db.Model):
    __tablename__ = "behavioral_logs"

    log_id = db.Column(db.Integer, primary_key=True)
    session_id = db.Column(db.Integer, db.ForeignKey("exam_sessions.session_id", ondelete="CASCADE"), nullable=False)
    event_type = db.Column(db.String(50), nullable=False)
    event_data = db.Column(db.JSON)
    logged_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
